### PR TITLE
add some additional message interpreters for debugging.

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add some additional debug messages for showing different message types.  Issue #1174, fix the unknown message 

### Proposed changes

- add messages for some additional command types
- make some performance tweaks for the actionMessage toByteArray operations

### CODE REVIEW

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ tests/helics/core/ActionMessage-tests.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5430845&bid=17808291)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ tests/helics/core/ActionMessage-tests.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5430845&bid=17808291)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ tests/helics/core/ActionMessage-tests.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5430845&bid=17808291)

**Comment by phlptp** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ tests/helics/core/ActionMessage-tests.cpp  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GMLC-TDC/HELICS/pullRequest?prid=5430845&bid=17808291)

File: src/helics/core/ActionMessage.cpp [hunk 1]
```diff
@@ -405,7 +417,7 @@ int ActionMessage::fromByteArray(const char* data, int buffer_size)
         payload.assign(data, sz);
         data += sz;
     }
-    int stringCount = *data;
+    int stringCount = static_cast<unsigned char>(*data);
```
Comment (nightlark) [inline src/helics/core/ActionMessage.cpp:420]: There's a max of 255 strings? Should this be a `uint8_t`?

Comment (phlptp) [reply src/helics/core/ActionMessage.cpp:420]: Yes there is a max of 255 strings.  And the variable needs to be an int for later comparison and use.  So either it gets cast here or later,   clang tidy complained about *data being a signed char so the recommendation was to cast to an unsigned char and then use the implicit conversion to int(which C++ does to everything anyways).

**Pull request merged**